api/projects/models.py

```diff
@@ -26,16 +29,46 @@
 from projects.managers import ProjectManager
 from projects.tasks import write_environments_to_dynamodb
 
+if typing.TYPE_CHECKING:
+    from environments.models import Environment
+    from features.models import Feature
+    from segments.models import Segment
+
+# TODO #2797 add audit log tests for permission changes
+
 project_segments_cache = caches[settings.PROJECT_SEGMENTS_CACHE_LOCATION]
 environment_cache = caches[settings.ENVIRONMENT_CACHE_NAME]
 
 
-class Project(LifecycleModelMixin, SoftDeleteExportableModel):
+UNAUDITED_PROJECT_FIELDS = (
+    "hide_disabled_flags",
+    "enable_dynamo_db",
+    "prevent_flag_defaults",
+    "enable_realtime_updates",
+    "only_allow_lower_case_feature_names",
+    "feature_name_regex",
+    "max_segments_allowed",
+    "max_features_allowed",
+    "max_segment_overrides_allowed",
+)
+
+
+class Project(
+    LifecycleModelMixin,
+    SoftDeleteExportableModel,
+    abstract_base_auditable_model_factory(
+        RelatedObjectType.PROJECT,
+        UNAUDITED_PROJECT_FIELDS,
+        audit_create=True,
+        audit_delete=True,
```

Comment:
What's the reasoning behind not auditing update here? (I don't think auditing the project at all was part of the scope so if that's the answer, that's cool, just checking)

Reply:
The spec was to audit project creation and deletion. Unlike other entities, it did not specify auditing updates. Possibly it should? See earlier discussion about fields.